src/init.ts

```diff
@@ -3,13 +3,10 @@ type PrimitiveValue = number | symbol | string | boolean | bigint | null | undef
 type PrimitiveOrPointer = Pointer | PrimitiveValue;
 export type ProxyTarget = CallableFunction | any[] | object;
 type ShadowTarget = CallableFunction | any[] | object;
-type ProxyTargetType = "object" | "function" | "array";
 type CallablePushTarget = (
     pointer: () => void,
-    typeofNextTarget: ProxyTargetType,
-    protoInNextTarget: boolean | undefined, // only for typeofTarget === 'function'
-    functionNameOfNextTarget: string | undefined, // only for typeofTarget === 'function'
-    isNextTargetAnArray: boolean | undefined, // only for typeofTarget !== 'function'
+    targetTraits: number,
```

Comment:
I have to use a more generic type here because TargetTraits needs to be defined inside the init function since that type declaration is required at runtime